### DISCUSSION TITLE
ripping without converting to flac possible?

### DISCUSSION THREAD

**Discussion by dschilde** [whipper-team/whipper#584] [Q&A]
Hello all,

on Arch-Linux, is it possible to just extract the audio data and save them to separate wav-files? I cannot find a hint on the man-page and want those files later to be converted to mp3 using lame encoder.

thanks

**Comment by martinwguy** [comment]
No, it looks like FLAC is compulsory.

**Comment by tombay** [comment]
If you really want to do this you can always edit the python file "whipper/program/flac.py" and add a line after check_call(['flac.
e.g. diff whipper/program/flac.py modified/whipper/program/flac.py
17a18

        check_call(['cp', infile, outfile.replace(".flac.part", ".wav")])

So this will copy the original .wav file (which whipper cdparanoia rips to) to the destination location.
A proper implementation would be to add an option to the rip process.
The location of the Whipper python source would be under the python package distribution tree (like /usr/local/lib/python3.12/dist-packages/wh..../whipper/program/flac.py).
Of course you would have to do this after every whipper upgrade (and maybe python) and probably need root privileges.

Alternatively you could use ffmpeg to convert the flac to wav and then use lame.
e.g. ffmpeg -i input.flac output.wav
You may need to handle encoding if your source is 24 bit:
ffmpeg -i input.flac -c:a pcm_s24le output.wav

see (https://superuser.com/questions/746969/ffmpeg-to-convert-from-flac-to-wav)
